unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotsView.cs

```diff
@@ -0,0 +1,62 @@
+using DCL.Helpers;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.Backpack
+{
+    public class AvatarSlotsView : BaseComponentView, IAvatarSlotsView
+    {
+        [SerializeField] private RectTransform avatarSlotsContainer;
+
+        [Header("Prefab references")]
+        [SerializeField] private GameObject avatarSlotSectionPrefab;
+        [SerializeField] private GameObject avatarSlotPrefab;
+        [SerializeField] private GameObject sectionSeparator;
+
+        private readonly Dictionary<string, Transform> avatarSlotSections = new ();
+        private readonly Dictionary<string, IAvatarSlotComponentView> avatarSlots = new ();
+
+        public event Action<string, bool> OnToggleAvatarSlot;
+
+        public void CreateAvatarSlotSection(string sectionName, bool addSeparator)
+        {
+            avatarSlotSections.Add(sectionName, Instantiate(avatarSlotSectionPrefab, avatarSlotsContainer).transform);
+
+            if (addSeparator)
+                Instantiate(sectionSeparator, avatarSlotsContainer);
+        }
+
+        public void RebuildLayout()
+        {
+            // Needed because adding elements to a transform that contains a
+            // layout does not refresh the child placements
+            Utils.ForceRebuildLayoutImmediate(avatarSlotsContainer);
+        }
+
+        public void AddSlotToSection(string sectionName, string slotCategory)
+        {
+            IAvatarSlotComponentView avatarSlot = Instantiate(avatarSlotPrefab, avatarSlotSections[sectionName]).GetComponent<IAvatarSlotComponentView>();
```

Comment:
I think we should use object pooling to manage the slots instantiations. Maybe it's not currently so important since we have a fixed list of items defined in a SO, but if it's going ot change in the future to be more dynamic, we should be prepared for that. Wdyt? (we can discuss it among the 3 of us and make a decision if you want).

Reply:
We can do it but when we add support in the future for dynamic avatars, now it's done on start and never changes, so the pool would be "pointless" and wouldn't add any improvements

Reply:
Ok, in this case there is no problem as it is going to be instantiated only at the beggining 👍